[awc1116] ip

awc1116 opened the pull request:
# Doopies 
>[“You don’t need a new plan for next year. You need a commitment.” – Seth Godin](https://dansilvestre.com/productivity-quotes/#4_%E2%80%9CYou_dont_need_a_new_plan_for_next_year_You_need_a_commitment%E2%80%9D_%E2%80%93_Seth_Godin)


### Doopies help you to note down your plans. It's
- Text-based
- Easy to learn
- ~~FASR~~ SUPER FAST to use


### All you need to do is,
1. Download it from [here](https://github.com/awc1116/ip/releases/tag/A-Release).
2. If double clicking on the downloaded file works, skip to step 6
3. Copy the jar file into an empty folder.
4. Open a command window in that folder.
5. Run the command `java -jar "{filename}.jar"` e.g. 'java - jar "doopies.jar"` (i.e., run the command in the same folder as the jar file).
6. Add your tasks.
7. Let it manage your tasks for you :grin:
> [!NOTE]
> The double quotes around the filename in the `java -jar "{filename}.jar"` is not normally needed, but it is needed if the filename contains special characters such as `spaces` or `[`.


### And it is **FREE**


### Features:
- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing reminders

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

awc1116 committed:
Change Duke to Doopies and added greetings

awc1116 committed:
added echo functionality

awc1116 committed:
added List package with MyList class

awc1116 committed:
added mark and unmark function

awc1116 committed:
rename myList to Notebook and added in the different tasks

awc1116 committed:
handle exceptions

awc1116 committed:
added delete function

awc1116 committed:
Automated Text UI Testing

awc1116 committed:
added save and load to store data in hard disk

awc1116 committed:
Merge branch 'branch-Level-7'

awc1116 committed:
added DateTime formatter

awc1116 committed:
Merge branch 'branch-Level-8'

awc1116 committed:
rearrange the various packages and classes

awc1116 committed:
Merge branch 'add-gradle-support'

awc1116 committed:
edited formatting and added gradle

awc1116 committed:
added JUnit tests

awc1116 committed:
added javaDocs comments

awc1116 committed:
correct coding style

awc1116 committed:
add in findCommand class, method find() in Notebook and relative test cases

awc1116 committed:
Merge branch 'branch-A-JavaDoc'

awc1116 committed:
Merge branch 'branch-A-CodingStandard'

awc1116 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Doopies.java
#	src/main/java/Doopies/command/DeadlineCommand.java
#	src/main/java/Doopies/command/EventCommand.java
#	src/main/java/Doopies/command/ToDoCommand.java
#	src/main/java/Doopies/notebook/Deadline.java
#	src/main/java/Doopies/notebook/Event.java
#	src/main/java/Doopies/notebook/Notebook.java

awc1116 committed:
edit changes that were messed up during the merge

awc1116 committed:
Edit changes that were messed up during the merge

awc1116 committed:
Merge branch 'master' of https://github.com/awc1116/ip

awc1116 committed:
Minor edits to code

1. Move class Doopies to doopies package.
2. Move class DoopiesTest to doopies package under Test folder.
3. Remove all javaDocs comments for private attributes, constructors and methods.

awc1116 committed:
Correct directory of Doopies class in build.java

change from mainClass.set("Doopies") to mainClass.set("doopies.Doopies")

awc1116 committed:
Check coding style for all files

1. Implemented checkstyle in gradle.build
2. Added config file with the 2 checkstyle files in it.
3. run the checkstyle test and edited accordingly.

SuspectBlue commented on src/main/java/Doopies/notebook/Task.java:
```diff
@@ -0,0 +1,92 @@
+ */
+public abstract class Task {
+    private final String task;
+    private final boolean done;
```
Consider renaming the done field to isDone to better align with Java boolean naming conventions

SuspectBlue commented on src/main/java/Doopies/Exception/IndexOutOfBoundException.java:
```diff
@@ -0,0 +1,8 @@
+package doopies.exception;
+
+public class IndexOutOfBoundException extends Exception {
+    public IndexOutOfBoundException(String message) {
+
+        super(message);
+    }
+}
```
I like the use of multiple exception types here allows for more precise error handling!

SuspectBlue commented on src/main/java/Doopies/util/Parser.java:
```diff
@@ -0,0 +1,81 @@
+        for (DateFormat format : DateFormat.values()) {
+            try {
+                return Optional.of(LocalDateTime.parse(dateStr, format.getFormatter()));
+            } catch (DateTimeParseException ignored) {
+            }
```
Consider adding logging or an informative message in your catch block

SuspectBlue reviewed (commented):
LGTM!
Just some minor nits.

darHH reviewed (commented):
Scanned your code for about 20 minutes and I have to say it is really well written! Very OOP-ed and Javadoc comments are very well formatted and comprehensive. Didnt find any specific errors (will probably have to look at it much more). Overall, good job and keep it up!

awc1116 committed:
Add GUI

1. Created Main.java, MainWindow.java, DialogBox.java in doopies.userinterface package for GUI implementation.
2. Shifted Doopies.java to doopies.userinterface package.
3. Edited Doopies.java and Ui.java to match the GUI inplementation.
4. Added resources file with images and view file in it for FXML and image items.

awc1116 committed:
Use java varargs

1. for method add(Task task) in Notebook class -> add(Task... task) allowing to add multiple tasks to notebook at once

awc1116 committed:
Merge branch 'branch-A-Varargs'

* branch-A-Varargs:
  Use java varargs

awc1116 committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add GUI

awc1116 committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Check coding style for all files

# Conflicts:
#	src/main/java/Doopies/Doopies.java
#	src/main/java/Doopies/userInterface/Ui.java
#	src/test/java/doopies/DoopiesTest.java

awc1116 committed:
Edit javaDocs comments and minor adjustments

1. Edited all javaDocs comments.
2. Edited Ui class.

awc1116 committed:
Add Into and Ending

1. When opened, the intro message will auto pop out
2. After bye command given, the bye message will be displayed and the app will be closed after 0.5sec

awc1116 committed:
Enhance GUI & Fix Scrolling in Doopies App

1.  UI Improvements: Added title bar with centered "Doopies" label and circular logo.
2. Scroll Pane Fixes: Enabled mouse wheel scrolling by simulating click + drag.

awc1116 committed:
Add javaDocs for showWelcome() under doopies.userinterface.MainWindow.java

awc1116 committed:
Add Assertions

1. add assertion for load method under Storage class to ensure that
filePath is not null
2. add assertion for parseCommand and parseMyDate method under Parser
to ensure that command and date string is not null or empty.

awc1116 committed:
Improve code quality

Created saveNotebook, parseInt and translate method in Command class,
so that the child class can use these methods instead of repeatedly
define or type in child class

awc1116 committed:
Add stream

Change find method in Notebook class to stream

awc1116 committed:
Merge pull request #1 from awc1116/branch-A-Assertions

Add Assertions

awc1116 committed:
Merge pull request #2 from awc1116/branch-A-CodeQuality

Improve code quality

awc1116 committed:
Merge pull request #3 from awc1116/branch-A-Streams

Add stream

awc1116 committed:
Add reminders functionality

Once you start the app, all not done (todo, deadline due within one day,
event starting within one day will be gathered and pop out as a reminder).
User can also type in the word reminders to get the reminders

1. Added a getTasksDueWithinOneDay method in Notebooks to get the reminders task.
2. Parser including a like from a "reminders" command from user.
3. Added a RemindersCommand to accept the "reminders" command and execute the necessary tasking.

awc1116 committed:
Add Help command

user can now key in help in the GUI to get all the commands for the
bot.

1) added HelpCommand class in doopies.command
2) include case in Paser class for help
3) added showHelp() function in UI class

awc1116 committed:
Change output date format

1. output date format from MMM dd YYYY, HH:mm a to
dd MMM YYYY, HH:mm a

awc1116 committed:
Edit README.md

edited README.md to fit my bot's user guide

awc1116 committed:
Rename User Guide and Update README.md

1) Rename the previous README in docs to DoopiesUserGuide
2) Edited the README in the main file

awc1116 committed:
Edit code format

1) edited minor code formatting and some test cases

awc1116 committed:
Add Ui.png

awc1116 committed:
Change product screenshot directory

awc1116 committed:
Create Launcher class

create Launcher class as entry point to the app.
edit build.gradle to make jar file works.

awc1116 committed:
Edit version of jar

change back to no version instead of 2.0 to prevent confusion when
releasing.
